### DISCUSSION TITLE
How to install and use a project template using dotnet CLI

### DISCUSSION THREAD

**Discussion by awa5114** [dotnet/templating#4765] [Template authoring] [answered]
**crossposting from [stackexchange][3]**

I recently made a [post][1] in which I was asking about how to disable implicit usings in Visual Studio 2022. From that post I understood a couple of things:

- This can only be done per project basis using the `.csproj` file by not including the `ImplicitUsings` tag
- It is possible to export a template of such a project and reuse it in other projects using `dotnet`.

Since I am interested in generating customized console applications directly from the command line (Powershell, to be specific), I first exported my adjusted console app as a template using the Visual Studio 2022 UI under *Project > Export Template > Project template*. This generated a `.zip` file which was dumped in a generic Templates folder in the Visual Studio installation directory.

According to the [link][2] in the reddit post, it should be possible to then make use of this template using `dotnet new --install` and then `dotnet new template`.

I have tried the following obvious commands:

`dotnet new <path_to_zip_template_file`
`dotnet new --install <path_to_zip_template_file`

The first command crashed with:

> No templates found matching `<path_to_zip_template_file>`

The second command crashed with:

> `<path_to_zip_template_file` is not supported

I'm probably doing something way off. Does anyone know what that might be? 

  [1]: https://www.reddit.com/r/csharp/comments/ut36r5/disabling_autoloading_of_default_namespaces_in/
  [2]: https://docs.microsoft.com/en-us/dotnet/core/tools/custom-templates
  [3]: https://superuser.com/questions/1721991/how-to-install-and-use-a-project-template-using-dotnet-cli

**Comment by sayedihashimi** [comment, marked as answer]
Hi @awa5114 the Export Template functionality in Visual Studio is using a different technology and doesn't work with `dotnet new`. I have guidance on how to create templates for `dotnet new` and Visual Studio at https://aka.ms/netcore-templates.

**Reply by awa5114** [reply]
Thanks, I was already familiar with that documentation and also watched a couple of your videos on youtube. Unfortunately it all seems primarily concerned with how to make a template from scratch, which is an overkill for what I'm trying to do. I simply need a way to start from an existing template (i.e. Console App), make a few minor changes to it and then make that available to `dotnet new`...

**Reply by sayedihashimi** [reply]
@awa5114 you don't need to start from scratch with a new project, just add the `.template.config` folder with the `template.json` file inside of it to your existing project folder.
